### PR TITLE
feat: 107 use video texture

### LINKED ISSUE #107: UseVideoTexture
### Description

To allow users to easily load video as a texture

### Suggested solution

https://github.com/pmndrs/drei#usevideotexture

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] I agree to follow this project's [Code of Conduct](https://github.com/Tresjs/cientos/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/Tresjs/cientos/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://cientos.tresjs.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/tresjs/cientos/issues) that reports the same bug to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by JaimeTorrealba**
closes #107

### CODE REVIEW

File: src/core/loaders/useVideoTexture.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    playsInline: true,
+    ...options,
+  }
+
+  function loadTexture(): Promise<VideoTexture> {
+    return new Promise(res => {
+      const video = Object.assign(document.createElement('video'), {
+        src: (typeof src === 'string' && src) || undefined,
+        crossOrigin,
+        loop,
```
Comment (alvarosabu) [inline src/core/loaders/useVideoTexture.ts:55]: @JaimeTorrealba please add error handling to the Promise, in case the src is wrong or could not be loaded

**Review by alvarosabu — CHANGES_REQUESTED** [review]
Looks amazing man, really nice work, just add the error handling to the loading of the video, and we  can merge

File: src/core/loaders/useVideoTexture.ts (hunk 1 above)
Comment (JaimeTorrealba) [reply src/core/loaders/useVideoTexture.ts:55]: Done Alvaro

**Review by alvarosabu — APPROVED** [review]

**Review by alvarosabu — CHANGES_REQUESTED** [review]

File: src/core/loaders/useVideoTexture.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        muted,
+        autoplay: true,
+        ...rest,
+      })
+      const texture = new VideoTexture(video)
+      video.addEventListener(unsuspend, () => res(texture))
+      video.addEventListener('error', () => rej())
```
[the remark this answers is not in the document]
Comment (JaimeTorrealba) [reply src/core/loaders/useVideoTexture.ts]: The error object is pretty much useless, so I'll keep the error log message

image

File: src/core/loaders/useVideoTexture.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    crossOrigin: 'Anonymous',
+    muted: true,
+    loop: true,
+    start: true,
+    playsInline: true,
+    ...options,
+  }
+
+  function loadTexture(): Promise<VideoTexture> {
+    return new Promise((res, rej) => {
```
[the remark this answers is not in the document]
Comment (alvarosabu) [reply src/core/loaders/useVideoTexture.ts]: @JaimeTorrealba please use `resolve` and `reject`. Avoid shorting variable and method names

**Review by alvarosabu — CHANGES_REQUESTED** [review]

**Review by alvarosabu — APPROVED** [review]

**Pull request merged**